src/besseli.jl

```diff
@@ -205,6 +206,34 @@ function _besseli(nu::Integer, x::T) where T <: Union{Float32, Float64}
     end
 end
 
+function _besseli(nu::AbstractRange, x::T) where T
+    (nu[1] >= 0 && step(nu) == 1) || throw(ArgumentError("nu must be >= 0 with step(nu)=1"))
+    len = length(nu)
+    isone(len) && return [besseli(nu[1], x)]
+    out = zeros(T, len)
+    k = len
+    inu = zero(T)
+    while abs(inu) < floatmin(T)
+        if besseli_underflow_check(nu[k], x)
+            inu = zero(T)
+        else
+            inu = _besseli(nu[k], x)
+        end
+        out[k] = inu
+        k -= 1
+        k < 1 && break
+    end
+    if k > 1
+        out[k] = _besseli(nu[k], x)
+        out[1:k+1] = besselk_down_recurrence!(out[1:k+1], x, nu[1:k+1])
```

Comment:
why are these copies needed?

Reply:
I was hoping that simply having
```
if k > 1
        out[k] = _besseli(nu[k], x)
        besselk_down_recurrence!(out[1:k+1], x, nu[1:k+1])
end
```
would work and do that operation in place but for some reason it was not making that copy appropriately with out explicitly having the `out[1:k+1]` statement....

Reply:
Ok I think this works...
```julia
if k > 1
            out[k] = _besselj(nu[k], x)
            tmp = @views out[1:k+1]
            besselj_down_recurrence!(tmp, x, nu[1:k+1])
            return out
        else
```
Looks like that gets rid of that extra allocation

Reply:
That seems a bit subtle what is going on there but.....
```julia
julia> @btime besselj(0:500, 20.0)
  7.620 μs (1 allocation: 4.06 KiB)
```
It does remove all of the unnecessary allocations. 

You think that is ok?

Reply:
much better.

Reply:
you could also write this with `@view` to remove a line, but this is good as is.

Reply:
Perfect - thank you! Was a bit stumped on properly using `@views` but this is very fast now.